geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlStatementFactory.java

```diff
@@ -17,38 +17,45 @@
 import java.util.List;
 
 class SqlStatementFactory {
+  private final String quote;
+
+  public SqlStatementFactory(String identifierQuoteString) {
+    this.quote = identifierQuoteString;
+  }
 
   String createSelectQueryString(String tableName, List<ColumnValue> columnList) {
     assert columnList.size() == 1;
     ColumnValue keyCV = columnList.get(0);
     assert keyCV.isKey();
-    return "SELECT * FROM \"" + tableName + "\" WHERE \"" + keyCV.getColumnName() + "\" = ?";
+    return "SELECT * FROM " + quote + tableName + quote + " WHERE " + quote + keyCV.getColumnName()
```

Comment:
we could add a helper method to quote around table/column name: 
String getQuotedTableColumnName(name) {
  return quote + name + quote;
}